core/src/main/java/org/infinispan/distribution/group/impl/GroupManagerImpl.java

```diff
@@ -62,7 +64,8 @@ else if (possibleMethods.size() == 1)
             throw new IllegalStateException(Util.formatString("Cannot define more that one @Group method for class hierarchy rooted at %s", clazz.getName()));
     }
 
-    @Inject private DistributionManager distributionManager;
+    @Inject private ComponentRegistry componentRegistry;
+    private DistributionManager distributionManager;
```

Comment:
Please move the comment about the dependency cycle here

Reply:
Sure